tests/Events.elm

```diff
@@ -0,0 +1,84 @@
+module Events exposing (..)
+
+import Expect
+import Html exposing (Html, button, div, text, input)
+import Html.Attributes as Attr exposing (href)
+import Html.Events exposing (onClick, onInput)
+import Html.Lazy as Lazy
+import Test exposing (..)
+import Test.Html.Events as Events
+import Test.Html.Query as Query exposing (Single)
+import Test.Html.Selector exposing (tag)
+
+
+all : Test
+all =
+    describe "trigerring events"
+        [ test "returns msg for click on element" <|
+            \() ->
+                Query.fromHtml sampleHtml
+                    |> Query.findAll [ tag "button" ]
+                    |> Query.first
+                    |> Events.msgFor "click" "{}"
+                    |> Expect.equal (Ok SampleMsg)
+        , test "returns msg for click on lazy html" <|
+            \() ->
+                Query.fromHtml sampleLazyHtml
+                    |> Query.findAll [ tag "button" ]
+                    |> Query.first
+                    |> Events.msgFor "click" "{}"
+                    |> Expect.equal (Ok SampleMsg)
+        , test "returns msg for click on mapped html" <|
+            \() ->
+                Query.fromHtml sampleMappedHtml
+                    |> Query.findAll [ tag "button" ]
+                    |> Query.first
+                    |> Events.msgFor "click" "{}"
+                    -- TODO: Html.Map is ignored when traversing the DOM, need to fix it to avoid repeated mapping on tests like this
```

Comment:
No it doesn't, I'm not sure how elm's vdom really work, but yeah, I though that when I got the element's function to decode events that it would already be mapped, but it is not, Html.Map seems to create a Node of some kind that we probably need to keep track of and call it to map our Msgs

But again, I'm still not very certain how all of this work, I prefer to open an issue and attempt to fix this later hahaha